straxen/contexts.py

```diff
@@ -261,52 +281,78 @@ def xenonnt_simulation(output_folder='./strax_data',
     for option in cmt_options:
-        st.config[option] = tuple(['cmt_run_id', _cmt_run_id_proc_only, *cmt_options[option]])
+        st.config[option] = ('cmt_run_id', cmt_run_id_proc, *cmt_options[option])
 
+    # Done with "default" usage, now to overwrites from file
+    #
     # Take fax config and put into context option
-    if overwrite_from_fax_file:
+    if overwrite_from_fax_file_proc or overwrite_from_fax_file_sim:
         fax_config = straxen.get_resource(fax_config, fmt='json')
-
         for fax_field, cmt_field in _config_overlap.items():
-            st.config[cmt_field] = tuple([cmt_options[cmt_field][0] + '_constant',
-                                          fax_config[fax_field]])
-
-    # Setup simulation
-    # Pass the CMT options to simulation (override all other config input methods)
-    else:
-        fax_config_override_from_cmt = dict()
-
-        for fax_field, cmt_field in _config_overlap.items():
-            fax_config_override_from_cmt[fax_field] = tuple(['cmt_run_id', cmt_run_id, *cmt_options[cmt_field]])
-        st.set_config({'fax_config_override_from_cmt': fax_config_override_from_cmt})
-
-    # Fix gain model
-    st.set_config({'gain_model_mc': tuple(['cmt_run_id', cmt_run_id, *cmt_options['gain_model']])})
-
-    # User customized overwrite
-    cmt_options = straxen.get_corrections.get_cmt_options(st)  # This includes gain_model_mc
-    for option in cmt_option_overwrite:
+            if overwrite_from_fax_file_proc:
+                st.config[cmt_field] = ( cmt_options[cmt_field][0] + '_constant',
+                                         fax_config[fax_field])
+            if overwrite_from_fax_file_sim:
+                st.config['fax_config_override_from_cmt'][fax_field] = (
+                         cmt_options[cmt_field][0] + '_constant',fax_config[fax_field])
+
+    # And as the last step - manual overrrides, since they have the highest priority
+    # User customized for simulation
+    for option in cmt_option_overwrite_sim:
         if option not in cmt_options:
             raise ValueError(f'Overwrite option {option} is not using CMT by default '
                              'you should just use set config')
+        if not option in _config_overlap.values():
+            raise ValueError(f'Overwrite option {option} does not have mapping from '
+                              'CMT to fax config! ')
+        for fax_key,cmt_key in _config_overlap.items():
+            if cmt_key==option:
+                continue
         _name_index = 2 if 'cmt_run_id' in cmt_options[option] else 0
-        st.config[option] = (cmt_options[option][_name_index] + '_constant', cmt_option_overwrite[option])
-
+        st.config['fax_config_override_from_cmt'][fax_key] = (
+                                    cmt_options[option][_name_index] + '_constant',
+                                    cmt_option_overwrite_sim[option])
+        del(fax_key,cmt_key,_name_index)
```

Comment:
I would remove this line. Why would you need to delete those parameters?

Reply:
it is a safety check, just in case it is accidentally used uninitialized later in the code